src/app/views/homepage/edit/EditHomepageItem.jsx

```diff
@@ -41,20 +94,268 @@ export default class EditHomepageItem extends Component {
+                console.error("error getting image details from image-api", error);
+                if (error.status == 401) {
+                    return;
+                }
+                notifications.add({
+                    type: "warning",
+                    message: "There was an error getting the image. You can still edit this item.",
+                    isDismissable: true
+                });
+            });
+    };
+
+    pollForUpdates = imageID => {
+        window.pollForUpdates = setInterval(() => {
+            this.getImage(imageID, false);
+        }, 1000);
+    };
+
+    stopPollingForUpdates = () => {
+        clearInterval(window.pollForUpdates);
+    };
+
+    getImageDownload = imageID => {
+        const downloadType = "original";
+        this.setState({ isGettingImage: true });
+        return image
+            .getDownloads(imageID, downloadType)
+            .then(response => {
+                const imageData = this.mapImageToState(response);
+                this.setState({ isGettingImage: false, imageData: imageData });
+            })
+            .catch(error => {
+                this.setState({ isGettingImage: false });
+                log.event(
+                    "error getting image downloads from image-api",
+                    log.data({ image_id: imageID, download_type: downloadType }),
+                    log.error(error)
+                );
+                console.error("error getting image downloads from image-api", error);
+                if (error.status == 401) {
+                    return;
+                }
+                notifications.add({
+                    type: "warning",
+                    message: "There was an error getting the image. You can still edit this item.",
+                    isDismissable: true
+                });
+            });
+    };
+
+    mapImageToState = image => {
+        try {
+            return {
+                url: image.href
+            };
+        } catch (error) {
+            log.event("error mapping image data to state", log.error(error));
+            // throw an error to let parent mapper catch and display notification
+            throw new Error(`Error mapping image data to state: \n ${error}`);
+        }
+    };
+
+    handleCancelClick = () => {
+        this.stopPollingForUpdates();
+        this.props.handleCancelClick();
+    };
+
+    handleRetryClick = () => {
+        this.setState({
+            imageState: "created",
+            upload: {}
+        });
+        this.bindInput();
+    };
+
+    handleRemoveImageClick = async () => {
+        this.setState({
+            image: "",
+            imageData: {
+                url: "",
+                title: "",
+                altText: ""
+            }
+        });
+        await this.createImageRecord();
+        this.bindInput();
+    };
+
+    renderImagePreview = () => {
```

Comment:
I think we should consider refactoring this method as it has multiple if statements. The various import/image fetching booleans only seem to be used in this method. Correct me if I'm wrong though

Could we maybe store the strings we return here as constants, and then in the various `setState` calls where the booleans are being set, we could instead have a single variable that is manipulated. Something like: `setState({ currentImageImportStatus: IMPORTING_IMAGE })`? 

That means in this method, we could then have a single if statement to check the image data object is there, and then a single `return <p>{this.state.currentImageStatus}</p>` beneath it. Let me know what you think